Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs

```diff
@@ -31,21 +31,35 @@ public AzureWebAppRoleEnvironmentTelemetryInitializer()
         }
 
         /// <summary>
-        /// Initializes <see cref="ITelemetry" /> device context.
+        /// Initializes <see cref="ITelemetry" /> device context and helps keep any heartbeat values in sync as well.
         /// </summary>
         /// <param name="telemetry">The telemetry to initialize.</param>
         public void Initialize(ITelemetry telemetry)
         {
+            string nodeName = string.Empty;
+
             if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleName))
             {
-                string name = LazyInitializer.EnsureInitialized(ref this.roleName, this.GetRoleName);
+                var name = LazyInitializer.EnsureInitialized(ref this.roleName, this.GetRoleName);
                 telemetry.Context.Cloud.RoleName = name;
             }
 
             if (string.IsNullOrEmpty(telemetry.Context.GetInternalContext().NodeName))
             {
-                string name = LazyInitializer.EnsureInitialized(ref this.nodeName, this.GetNodeName);                
-                telemetry.Context.GetInternalContext().NodeName = name;
+                nodeName = LazyInitializer.EnsureInitialized(ref this.nodeName, this.GetNodeName);
+                telemetry.Context.GetInternalContext().NodeName = nodeName;
+            }
+
+            // ensure heartbeat values are up to date...
+            if (string.IsNullOrEmpty(this.lastNodeValue))
+            {
+                this.lastNodeValue = nodeName;
+            }
+            else if (!nodeName.Equals(this.lastNodeValue, StringComparison.Ordinal))
+            {
+                // if the AppServices heartbeat telemetry module exists, signal it to update the values in heartbeat
+                AppServicesHeartbeatTelemetryModule.Instance?.UpdateHeartbeatWithAppServiceEnvVarValues();
```

Comment:
@SergeyKanzhelev, here is where I signal the `AppServicesHeartbeatTelemetryModule` to update the values in the heartbeat properties. This is the reason I chose to go with the Singleton pattern. Perhaps there is a better way to go about this?